browser-test/src/session_timeout.test.ts

```diff
@@ -0,0 +1,73 @@
+import {test, expect} from './support/civiform_fixtures'
+import {enableFeatureFlag, loginAsAdmin, loginAsTestUser} from './support'
+import {SessionTimeout} from './support/session_timeout'
+
+test.describe('Session timeout warnings for Admin', () => {
+  test.beforeEach(async ({page}) => {
+    await enableFeatureFlag(page, 'session_timeout_enabled')
+    await loginAsAdmin(page)
+  })
+
+  test('shows inactivity warning modal, extends session, show session length warning modal and logout', async ({
+    page,
+  }) => {
+    const sessionTimeout = new SessionTimeout(page)
+    await sessionTimeout.verifyTimeoutModalBehavior(
+      'admin-session-inactivity-warning-modal',
+      'admin-session-length-warning-modal',
+    )
+    await sessionTimeout.verifyAdminSessionLengthDialogLogout()
+  })
+
+  test('dismisses session length warning modal when Cancel button is clicked', async ({
+    page,
+  }) => {
+    const sessionTimeout = new SessionTimeout(page)
+    await sessionTimeout.verifyModalDismissal()
+
+    await test.step('Advance to timeout and verify logout', async () => {
+      await sessionTimeout.advanceToSessionTimeout()
+      await page.waitForURL((url) => url.pathname === '/programs')
+      await expect(page.locator('a[href="/applicantLogin"]')).toHaveText(
+        'Log in',
+      )
+    })
+  })
+})
+
+function defineApplicantTests(options: {isNorthStar?: boolean} = {}) {
+  const description: string = options.isNorthStar
+    ? 'with NorthStar UI'
+    : 'Session timeout warnings for Applicants'
+  const config = options.isNorthStar ? {tag: ['@northstar']} : {}
+
+  test.describe(`${description}`, config, () => {
```

Comment:
(blocking) There's no need to wrap this in a function. It'll make removing it later when pre-northstar goes away more difficult.